commands/addPermissions.ts

```diff
@@ -0,0 +1,66 @@
+import { ApplicationCommandPermissionData, CommandInteraction } from 'discord.js';
+import { SlashCommandBuilder, roleMention } from '@discordjs/builders';
+import path from 'path';
+import { writeFile } from 'fs';
+import { ICommand } from '../types/command';
+
+const dirname = path.resolve();
+const pathToJson = path.resolve(dirname, '../customPermissions.json');
+const file = await import(pathToJson);
+
+export const addPermissions: ICommand = {
+  config: new SlashCommandBuilder()
+    .setName('addpermissions')
+    .setDescription('Add permissions to command')
+    .setDefaultPermission(false)
+    .addRoleOption((option) => option.setName('role').setDescription('The role to add permission').setRequired(true))
+    .addStringOption((option) =>
+      option.setName('command').setDescription('Name of the command to update permissions').setRequired(true)
+    ),
+  execute: async (interaction: CommandInteraction) => {
+    await interaction.deferReply();
+    const { guild, options } = interaction;
+    const roleID = options.getRole('role').id;
+    const commandName = options.getString('command');
+
+    const writeToFile = (commandId: string, permissions: Array<ApplicationCommandPermissionData>) => {
+      if (commandId in file) {
+        file[commandId].permissions = file[commandId].permissions.concat(permissions);
+      } else {
+        file[commandId] = {
+          id: commandId,
+          permissions
+        };
+      }
+
+      const fileString = JSON.stringify(file);
+      writeFile(pathToJson, fileString, (err) => {
```

Comment:
is `writeFile` an async operation? If so, `writeToFile` might need to be an aync function